### ISSUE TITLE
`dict` literal not accepted as `Mapping` argument

### ISSUE THREAD

**Issue by tibdex** [astral-sh/ty#1975]
### Summary

In:

```python
from collections.abc import Mapping
from typing import Literal

type DataType = Literal["BOOL", "INT", "FLOAT"]

def process(data_types: Mapping[str, DataType], /) -> None:
    ...

process({"foo":"BOOL"})
```

I would expect the type checker to see that `{"foo":"BOOL"}` conforms to `Mapping[str, DataType]` when it is declared, and since it is not assigned anywhere, it cannot be mutated in a way that would modify its type before being passed as an argument.

However, I get:

```
Argument to function `process` is incorrect: Expected `Mapping[str, DataType]`, found `dict[Unknown | str, Unknown | str]` (invalid-argument-type) [Ln 9, Col 9]
```

[TypeScript 5.9.3 behaves like I expect](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAIghsOAVc0C8UBEAhA8rgGUygB8sBJAOSWLMwDEDcBBGgKDYGMB7AOwGdgUMACdunCP35QMACgAmCZKn4AuKACUIceXwA2IADwBvANq84AWwirBIgJa8A5gF118RCkgBfAHwBKGV8oY28Abg5RcUl+WWNuAGt1HHwib382KIkpON5uYFwkrBhWAFFMdI4eASF+bmtcACMAKwhOIQx4opTCCoismNk6hpa24H8woA):

```typescript
type DataType = "BOOL" | "INT" | "FLOAT"

const process = (dataTypes: Readonly<{[name:string]: DataType}>) => {};

// This is accepted, but ty rejects it :(
process({ok: "BOOL"})

// Type '"DATE"' is not assignable to type 'DataType'
process({notOk: "DATE"})

const someObject = {ok: "BOOL"};
// Argument of type '{ ok: string; }' is not assignable to parameter of type // 'Readonly<{ [name: string]: DataType; }>'.
//   Property 'ok' is incompatible with index signature.
//    Type 'string' is not assignable to type 'DataType'.
process(someObject);
```

### Version

0.0.2.

**Comment by AlexWaygood** [comment]
Thanks! Please see #1576. We're working on solutions to this :-)

**Comment by tibdex** [comment]
Thanks
